src/monitors/monitor.py

```
@@ -0,0 +1,92 @@
+        self._buffer.append(data)
+        self._full.notify()
+        self._mutex.release()
+
+    def read(self):
+        data = None
+        self._mutex.acquire()
+        if len(self._buffer) == 0 and self._producers_online > 0:
+            self._full.wait()
+        try:
+            data = self._buffer.pop()
+        except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)